### PR TITLE
Mechanisms to prevent client identification

### PR DESCRIPTION
**Pull request by tireddy2**

### CODE REVIEW

File: draft-rdb-ohai-feedback-to-proxy-05.xml [hunk 1]
```diff
@@ -0,0 +1,848 @@
...
+        reshape traffic from a specific client or from all the clients
+        according to the received Oblivious Relay Feedback can experience
+        different levels of service denial by the Oblivious gateway and target
+        resources. There is no explicit mechanism for an Oblivious relay to
+        indicate to the server that the rate-limit information was processed
+        or was ignored.</t>
+      </section>
+    </section>
+
+    <section anchor="mechanism" title="Prevent client identification">
```
Comment (boucadair) [inline draft-rdb-ohai-feedback-to-proxy-05.xml]:
```suggestion
    <section anchor="mechanism" title="Prevent Client identification">
```

File: draft-rdb-ohai-feedback-to-proxy-05.xml [hunk 2]
```diff
@@ -0,0 +1,848 @@
...
+      <t>The following mechanisms can be used by the Oblivious relay to
```
Comment (boucadair) [inline draft-rdb-ohai-feedback-to-proxy-05.xml]:
```suggestion
      <t>The following mechanisms can be used by an Oblivious relay to
```

File: draft-rdb-ohai-feedback-to-proxy-05.xml [hunk 3]
```diff
@@ -0,0 +1,848 @@
...
+      process "ohttp-target" parameter with value set to 2 :</t>
```
Comment (boucadair) [inline draft-rdb-ohai-feedback-to-proxy-05.xml]:
```suggestion
      process "ohttp-target" parameter with a value set to '2':</t>
```

File: draft-rdb-ohai-feedback-to-proxy-05.xml [hunk 4]
```diff
@@ -0,0 +1,848 @@
...
+
+      <t><list style="numbers">
+          <t>The Oblivious relay does not immediately act to rate-limit the
+          traffic from the client but starts maintaining a count of responses
+          to the client with "ohttp-target" parameter value set to "2" marked
+          as "potential malicious requests" and responses without the
+          parameter marked as "legitimate requests". The Oblivious relay can
+          rate-limit requests from the offending client for a certain duration
```
Comment (boucadair) [inline draft-rdb-ohai-feedback-to-proxy-05.xml]:
```suggestion
          rate-limit requests from the offending client for a certain duration (that is implementation-specific)
```

File: draft-rdb-ohai-feedback-to-proxy-05.xml [hunk 5]
```diff
@@ -0,0 +1,848 @@
...
+          <t>The Oblivious relay enforces the delayed rate-limiting of the
+          offending client discussed in step 1 provided a large number of
+          clients are actively using the relay and the total number of benign
+          clients is significantly a large number compared to offending
+          clients (e.g., total clients is greater than hundred thousand and
+          benign clients is above eighty percent of the total clients). This
+          way the anonymity set of benign clients will be a relatively larger
+          number even after rate-limiting the offending clients.</t>
+
+          <t>The Oblivious relay acts to rate-limit the client provided it has
```
Comment (boucadair) [inline draft-rdb-ohai-feedback-to-proxy-05.xml]:
```suggestion
          <t>The Oblivious relay acts to rate-limit the traffic of the client provided it has
```

File: draft-rdb-ohai-feedback-to-proxy-05.xml [hunk 6]
```diff
@@ -0,0 +1,848 @@
...
+          a bad reputation score (Many security network security vendors offer
```
Comment (boucadair) [inline draft-rdb-ohai-feedback-to-proxy-05.xml]:
```suggestion
          a bad reputation score (e.g., many security network security vendors offer
```

File: draft-rdb-ohai-feedback-to-proxy-05.xml [hunk 7]
```diff
@@ -0,0 +1,848 @@
...
+          client reputation score, for instance, see <xref
+          target="Reputation"></xref>).</t>
+
+          <t>The Oblivious relay provider hosts a honeypot of clients to send
+          good traffic to the server. The relay checks if it receives the
+          ohttp-target parameter set to 2 for these benign clients to identify
+          if the target is misbehaving. For example, if the target is is
```
Comment (boucadair) [inline draft-rdb-ohai-feedback-to-proxy-05.xml]:
```suggestion
          if the target is misbehaving. For example, if the target is
```

File: draft-rdb-ohai-feedback-to-proxy-05.xml [hunk 8]
```diff
@@ -0,0 +1,848 @@
...
+          number even after rate-limiting the offending clients.</t>
+
+          <t>The Oblivious relay acts to rate-limit the client provided it has
+          a bad reputation score (Many security network security vendors offer
+          client reputation score, for instance, see <xref
+          target="Reputation"></xref>).</t>
+
+          <t>The Oblivious relay provider hosts a honeypot of clients to send
+          good traffic to the server. The relay checks if it receives the
+          ohttp-target parameter set to 2 for these benign clients to identify
```
Comment (boucadair) [inline draft-rdb-ohai-feedback-to-proxy-05.xml]:
```suggestion
          "ohttp-target" parameter set to '2' for these benign clients to identify
```

**Review by boucadair — CHANGES_REQUESTED** [review]
very minor edits

**Review by danwing — APPROVED** [review]

**Pull request merged**